src/components/ProductCard/ProductCard.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+import CartIcon from '@assets/icons/shopping_cart_01.svg?react';
+
+import type { Product } from '@/interfaces/Product';
+
+import headerStyles from '../../components/HeaderComponent/header.module.css';
+import styles from './ProductCard.module.css';
+
+interface ProductCardProps {
+    productData: Product;
+    onAddToCart: (item: Product) => void;
```

Comment:
you are repeat same type in props several times, create a separate type for it
```suggestion
    type AddToCartHandler = (item: Product) => void;
```